src/bundle/DependencyInjection/Compiler/NavigationHubPass.php

```diff
@@ -24,14 +24,23 @@ public function process(ContainerBuilder $container)
 
     private function processNavigationHubTag(ContainerBuilder $container, $tag, $index)
     {
-        $items = [];
+        $unsorted = [];
         $services = $container->findTaggedServiceIds($tag);
-        foreach (array_keys($services) as $serviceId) {
-            $items[] = new Reference($serviceId);
+        foreach ($services as $serviceId => $tags) {
+            foreach ($tags as $tag) {
+                $priority = isset($tag['priority']) ? $tag['priority'] : 0;
```

Comment:
Making default value reduces effort required to make extensions, while still gives an option to personalise it as you like. I would opt for a compromise here and leave default value, but still explicitly declare it for our services to make extending predictable.